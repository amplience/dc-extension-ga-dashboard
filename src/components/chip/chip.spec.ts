import { render } from '@testing-library/svelte';
import Chip from './chip.svelte';

describe('Chip', () => {
  it('should render a Chip', () => {
    const { container } = render(Chip, {
      label: 'test-label',
      onClick: () => {
        /* noop */
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an active Chip', () => {
    const { container } = render(Chip, {
      label: 'test-label',
      active: true,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
