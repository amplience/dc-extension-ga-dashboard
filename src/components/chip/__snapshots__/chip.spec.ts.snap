// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render a Chip 1`] = `
<div>
  <div
    class="   svelte-dfh104"
    data-testid="chip"
  >
    <span
      class="label svelte-dfh104"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`Chip should render a clickable Chip 1`] = `
<div>
  <div
    class=" clickable  svelte-dfh104"
    data-testid="chip"
  >
    <span
      class="label svelte-dfh104"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`Chip should render an active Chip 1`] = `
<div>
  <div
    class="active   svelte-dfh104"
    data-testid="chip"
  >
    <span
      class="label svelte-dfh104"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`Chip should render an clickable active Chip 1`] = `
<div>
  <div
    class="active clickable  svelte-dfh104"
    data-testid="chip"
  >
    <span
      class="label svelte-dfh104"
    >
      test-label
    </span>
     
  </div>
</div>
`;
